plugin-server/src/worker/ingestion/person-state.ts

```diff
@@ -137,45 +136,50 @@ export class PersonState {
-            const propertiesOnce = excludeProperties ? {} : this.eventProperties['$set_once'] || {}
-            // Catch race condition where in between getting and creating, another request already created this user
-            try {
-                const person = await this.createPerson(
-                    this.timestamp,
-                    properties || {},
-                    propertiesOnce || {},
-                    this.teamId,
-                    null,
-                    // :NOTE: This should never be set in this branch, but adding this for logical consistency
-                    this.updateIsIdentified,
-                    this.newUuid,
-                    this.event.uuid,
-                    [this.distinctId]
-                )
-                // :TRICKY: Avoid subsequent queries re-fetching person
-                this.personContainer = this.personContainer.with(person)
-                return true
-            } catch (error) {
-                if (!error.message || !error.message.includes('duplicate key value violates unique constraint')) {
-                    Sentry.captureException(error, {
-                        extra: {
-                            teamId: this.teamId,
-                            distinctId: this.distinctId,
-                            timestamp: this.timestamp,
-                            personUuid: this.newUuid,
-                        },
-                    })
-                }
-            }
+        const properties = excludeProperties ? {} : this.eventProperties['$set'] || {}
+        const propertiesOnce = excludeProperties ? {} : this.eventProperties['$set_once'] || {}
+
+        try {
+            return promiseRetry(() => this.createOrGetPerson(properties, propertiesOnce))
+        } catch (error) {
+            // TODO: figure out if we need to add Sentry here or it's already upstream
+            Sentry.captureException(error, {
+                extra: {
+                    teamId: this.teamId,
+                    distinctId: this.distinctId,
+                    timestamp: this.timestamp,
+                    personUuid: this.newUuid,
+                },
+            })
+            throw error
         }
+    }
 
-        // Person was likely created in-between start-of-processing and now, so ensure that subsequent queries
-        // to fetch person still return the right `person`
-        this.personContainer = this.personContainer.reset()
-        return false
+    // If we created the person we already updated the properties, in that case we return true
+    // If the person already existed we do nothing here, which means properties aren't updated and we return false
+    private async createOrGetPerson(properties: Properties, propertiesOnce: Properties): Promise<boolean> {
+        // getPersonId hits Redis cache first, if not there goes to Postgres
+        const personId = await this.db.getPersonId(this.teamId, this.distinctId)
+        if (personId) {
```

Comment:
Could there be a `person` with `Id` = 0? :sweat_smile:

Reply:
nope. we haven't specially added any id=0 persons and the id field by default starts from 1